hazelcast/src/main/java/com/hazelcast/config/AtomicLongConfig.java

```diff
@@ -46,6 +53,47 @@ public int getId() {
         return ConfigDataSerializerHook.ATOMIC_LONG_CONFIG;
     }
 
+    @Override
```

Comment:
Duplication is (still) evil. :)

Reply:
With serialization it's always better to have it in a class which is serialized. Hierarchies in serialization are tricky to handle. So I know David did a great job with non-duplication, but it's such a basic logic, so it's kind of pragmatic to handle it where it really happens.

Reply:
I copied this de-duplication approach from the `CollectionConfig`, which is extended to `SetConfig` and `ListConfig` (and apparently didn't cause any troubles with the serialization so far).

Reply:
I see the `AbstractBasicConfig implements IdentifiedDataSerializable` and even if the child classes don't bring any new fields, the serialization related method are in the children implementations.

Similar concern is for `equals` and `hashCode`.

This adds about 45 lines of unnecessary code to the codebase.

Reply:
`equals()` and `hashCode()` are a bit difficult, since we use a library to test them automatically, which has some issues when the methods with inheritance on these methods (and if they are not final etc.). I think I couldn't manage to de-duplicate them, if I remember correctly.

Reply:
I don't care how many lines to the codebase it adds. Simplicity is not only about the length of a class, but also about separation of concerns. This de-duplication is simply art for art's sake, not adding any added value. I want the serialization and equality to be a concern of the child class, because experience has proven that this is much simpler to maintain.